test/data/EnumerableMap.ts

```diff
@@ -0,0 +1,415 @@
+      const addressFive = bnToAddress(BigNumber.from('500'));
+      const addressSix = bnToAddress(BigNumber.from('600'));
+
+      describe('#at(uint256)', () => {
+        it('returns value coresponding to index provided', async () => {
+          await instance['set(address,address)'](addressOne, addressFour);
+          await instance['set(address,address)'](addressTwo, addressFive);
+          await instance['set(address,address)'](addressThree, addressSix);
+
+          expect(await instance['at(uint256)'](0)).to.deep.equal([
+            addressOne,
+            addressFour,
+          ]);
+          expect(await instance['at(uint256)'](1)).to.deep.equal([
+            addressTwo,
+            addressFive,
+          ]);
+          expect(await instance['at(uint256)'](2)).to.deep.equal([
+            addressThree,
+            addressSix,
+          ]);
+        });
+
+        describe('reverts if', () => {
+          it('index is out of bounds', async () => {
+            await expect(
+              instance['at(uint256)'](0),
+            ).to.be.revertedWithCustomError(
+              instance,
+              'EnumerableMap__IndexOutOfBounds',
+            );
+          });
+        });
+      });
+
+      describe('#contains(address)', () => {
+        it('returns true if value has been added', async () => {
+          await instance['set(address,address)'](addressOne, addressFour);
+          await instance['set(address,address)'](addressTwo, addressFive);
+          await instance['set(address,address)'](addressThree, addressSix);
+
+          expect(await instance['contains(address)'](addressOne)).to.be.true;
+          expect(await instance['contains(address)'](addressTwo)).to.be.true;
+          expect(await instance['contains(address)'](addressThree)).to.be.true;
+        });
+
+        it('returns false if value has not been added', async () => {
+          expect(await instance['contains(address)'](addressFour)).to.be.false;
+        });
+      });
+
+      describe('#length()', () => {
+        it('returns length of enumerable map', async () => {
+          expect(await instance['length()']()).to.equal(0);
+
+          await instance['set(address,address)'](addressOne, addressFour);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['set(address,address)'](addressTwo, addressFive);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['set(address,address)'](addressThree, addressSix);
+          expect(await instance['length()']()).to.equal(3);
+
+          await instance['remove(address)'](addressThree);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['remove(address)'](addressTwo);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['remove(address)'](addressOne);
+          expect(await instance['length()']()).to.equal(0);
```

Comment:
This is a good case for making multiple calls.  I would probably put them in a loop, though doing that is dangerous because if a loop has 0 iterations, the issue won't show up as a test failure.